lib/ui/common/ViewModelProviderBuilder.dart

```diff
@@ -39,10 +40,11 @@ class ViewModelProviderBuilder<T> extends StatelessWidget {
   Widget build(BuildContext context) {
     final errorBuilder = _errorBuilder ?? _defaultErrorBuilder;
     final loadingBuilder = _loadingBuilder ?? _defaultLoadingBuilder;
-
+    final provider = _defaultProvider ?? Provider.of<ViewModelProviderInterface<T>>(context, listen: false,);
```

Comment:
Here we use the "provider" lib to provide the thing that will be providing the viewModels and their changes basically the Bloc.
(need naming help here)